compiler/passes/docs.cpp

```diff
@@ -221,45 +246,50 @@ void createDocsFileFolders(std::string filename) {
  * should be placed.
  */
 std::string generateSphinxProject(std::string dirpath) {
-  // FIXME: This ought to be done in a TMPDIR, unless --save-rst is
-  //        provided... (thomasvandoren, 2015-01-29)
-
   // Create the output dir under the docs output dir.
-  const char * htmldir = astr(dirpath.c_str(), "/html");
+  const char * sphinxDir = dirpath.c_str();
 
   // Ensure output directory exists.
-  const char * mkdirCmd = astr("mkdir -p ", htmldir);
+  const char * mkdirCmd = astr("mkdir -p ", sphinxDir);
   mysystem(mkdirCmd, "creating docs output dir");
 
   // Copy the sphinx template into the output dir.
   const char * sphinxTemplate = astr(CHPL_HOME, "/third-party/chpldoc-venv/chpldoc-sphinx-project/*");
-  const char * cmd = astr("cp -r ", sphinxTemplate, " ", htmldir, "/");
+  const char * cmd = astr("cp -r ", sphinxTemplate, " ", sphinxDir, "/");
   mysystem(cmd, "copying chpldoc sphinx template");
 
-  const char * moddir = astr(htmldir, "/source/modules");
+  const char * moddir = astr(sphinxDir, "/source/modules");
   return std::string(moddir);
 }
 
 /* Call `make html` from inside sphinx project. */
-void generateSphinxOutput(std::string dirpath) {
-  const char * htmldir = astr(dirpath.c_str(), "/html");
-
-  // The virtualenv activate script is at:
-  //   $CHPL_HOME/third-party/chpldoc-venv/install/$CHPL_TARGET_PLATFORM/chpldoc-virtualenv/bin/activate
-  const char * activate = astr(
+void generateSphinxOutput(std::string sphinxDir, std::string outputDir) {
+  // The virtualenv active and sphinx-build scripts are in:
+  //   $CHPL_HOME/third-party/chpldoc-venv/install/$CHPL_TARGET_PLATFORM/chpldoc-virtualenv/bin/
+  const char * venvBinDir = astr(
     CHPL_HOME, "/third-party/chpldoc-venv/install/",
-    CHPL_TARGET_PLATFORM, "/chpdoc-virtualenv/bin/activate");
-
-  // Run: `. $activate && cd $htmldir && $CHPL_MAKE html`
+    CHPL_TARGET_PLATFORM, "/chpdoc-virtualenv/bin/");
+  const char * activate = astr(venvBinDir, "activate");
+  const char * sphinxBuild = astr(venvBinDir, "sphinx-build");
+
+  // Run:
+  //   . $activate &&
+  //     sphinx-build -b html
+  //     -d $sphinxDir/build/doctrees -W
+  //     $sphinxDir/source $outputDir
+  const char * cmdPrefix = astr(". ", activate, " && ");
   const char * cmd = astr(
-    ". ", activate,
-    " && cd ", htmldir, " && ",
-    CHPL_MAKE, " html");
+    cmdPrefix,
+    sphinxBuild, " -b html -d ",
+    sphinxDir.c_str(), "/build/doctrees -W ",
+    sphinxDir.c_str(), "/source ", outputDir.c_str());
   mysystem(cmd, "building html output from chpldoc sphinx project");
+  printf("HTML files are at: %s\n", outputDir.c_str());
+  printf("Begin by opening index.html in a browser.\n");
```

Comment:
I'm not crazy about this messaging. Do you think we need to call out the index.html file? Should the outputDir be the absolute path instead of the relative one?